JavaDoc error during compilation

A JavaDoc error occurs when compiling WebFlow with Java 11:
```
[ERROR] MavenReportException: Error while generating Javadoc:
Exit code: 1 - /Users/romanstrobl/Lime-dev/powerauth-webflow/powerauth-tpp-engine-model/src/main/java/io/getlime/security/powerauth/app/tppengine/model/certificate/ICACertificateParser.java:24: error: package sun.security.x509 is not visible
import sun.security.x509.AVA;
```

Additional error:
```
[ERROR] Error fetching link: /Users/romanstrobl/Lime-dev/powerauth-webflow/powerauth-tpp-engine-model/target/apidocs. Ignored it.
```

--- a/docs/Compilation,-Packaging-and-Deployment.md
+++ b/docs/Compilation,-Packaging-and-Deployment.md
@@ -1,6 +1,6 @@
 # Compilation, Packaging and Deployment
 
-Web Flow uses Maven for compilation and packaging. Java 8 should be used to compile Web Flow.
+Web Flow uses Maven for compilation and packaging. Java 11 should be used to compile Web Flow.
 
 Web Flow can be deployed to any Java web container (such as Tomcat) using war artifacts produced during compilation.
 

--- a/pom.xml
+++ b/pom.xml
@@ -89,8 +89,8 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <maven.compiler.source>1.8</maven.compiler.source>
-        <maven.compiler.target>1.8</maven.compiler.target>
+        <maven.compiler.source>11</maven.compiler.source>
+        <maven.compiler.target>11</maven.compiler.target>
         <maven-jar-plugin.version>3.2.0</maven-jar-plugin.version>
         <maven-javadoc-plugin.version>3.2.0</maven-javadoc-plugin.version>
         <maven-war-plugin.version>3.2.2</maven-war-plugin.version>
@@ -128,6 +128,7 @@
                 <version>${maven-javadoc-plugin.version}</version>
                 <configuration>
                     <failOnError>false</failOnError>
+                    <detectOfflineLinks>false</detectOfflineLinks>
                 </configuration>
                 <executions>
                     <execution>

--- a/powerauth-tpp-engine-model/pom.xml
+++ b/powerauth-tpp-engine-model/pom.xml
@@ -56,6 +56,21 @@
         <plugins>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>${maven-compiler-plugin.version}</version>
+                <configuration>
+                    <source>11</source>
+                    <target>11</target>
+                    <compilerArgs>
+                        <arg>--add-exports</arg>
+                        <arg>java.base/sun.security.util=ALL-UNNAMED</arg>
+                        <arg>--add-exports</arg>
+                        <arg>java.base/sun.security.x509=ALL-UNNAMED</arg>
+                    </compilerArgs>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
                 <version>${maven-javadoc-plugin.version}</version>
                 <configuration>

--- a/powerauth-tpp-engine-model/pom.xml
+++ b/powerauth-tpp-engine-model/pom.xml
@@ -52,4 +52,30 @@
         </dependency>
     </dependencies>
 
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-javadoc-plugin</artifactId>
+                <version>${maven-javadoc-plugin.version}</version>
+                <configuration>
+                    <failOnError>false</failOnError>
+                    <detectOfflineLinks>false</detectOfflineLinks>
+                    <additionalOptions>
+                        <option>--add-exports</option>
+                        <option>java.base/sun.security.x509=ALL-UNNAMED</option>
+                    </additionalOptions>
+                </configuration>
+                <executions>
+                    <execution>
+                        <id>attach-javadocs</id>
+                        <goals>
+                            <goal>jar</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+
 </project>

--- a/powerauth-webflow/pom.xml
+++ b/powerauth-webflow/pom.xml
@@ -19,7 +19,7 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <java.version>1.8</java.version>
+        <java.version>11</java.version>
     </properties>
 
     <dependencies>

